modules/tensor_mechanics/doc/content/modules/tensor_mechanics/ShellElements.md

```diff
@@ -0,0 +1,78 @@
+
+!media media/tensor_mechanics/shell_representation.png
+      style=width:50%;margin-left:2%;float:right
+      id=fig_shell
+      caption=Shell element with 4 nodes and 3 translational and 2 rotational degrees of freedom at each node.
+
+Each of the four nodes have 5 degrees of freedom - 3 displacements ($u_1^k$,$u_2^k$, $u_3^k$) and 2 rotations ($\alpha_k$ and $\beta_k$). The thickness ($a_k$) is assumed to be constant across the shell element and the initial normal ($^oV_n^k) at all nodes are same as the normal to the shell element, which is automatically computed using the coordinates of the nodes.
+
+The basic equation of motion for a quasi-static shell is the same as that of a continuum brick element:
+
+\begin{equation}
+\nabla \cdot \sigma = F_{ext}
+\end{equation}
+
+The quasi-static shell element implementation is Moose is spread across four different objects: strain computation ([ADComputeIncrementalShellStrain](/ADComputeIncrementalShellStrain.md) & [ADComputeFiniteShellStrain](/ADComputeFiniteShellStrain.md)), elasticity tensor computation ([ADComputeIsotropicElasticityTensorShell](/ADComputeIsotropicElasticityTensorShell.md)), stress computation ([ADComputeShellStress](/ADComputeShellStress.md)) and stress divergence kernel ([ADStressDivergenceShell](/ADStressDivergenceShell.md)).
+
+
+## Strain Calculation
+
+The first step to calculating stresses is to calculate the strains in the shell element. All the computations are performed in the natural coordinate system of the shell ($r_1$, $r_2$ and $r_3$). The geometry of the element at any time t is defined as:
+
+\begin{equation}
+^t x_i =  \sum_{k=1}^4 h_k ^t {x_i}^k + \frac{r_3}{2} \sum_{k=1}^4 a_k h_k ^tV_{ni}^k
+\end{equation}
+
+Therefore, the incremental displacements of any point with natural coordinate $r_i$ of the shell element at time $t$ in the global cartesian coordinate system ($^ox_1$, $^ox_2$ and $^ox_3$) are:
+
+\begin{equation}
+u_i = \sum_{k=1}^4 h_k u_i^k + \frac{r_3}{2} \sum_{k=1}^4 a_k h_k (-^tV_{2i}^k \alpha_k + ^tV_{1i}^k \beta_k)
+\end{equation}
+
+If $^t \mathbf{g_i} = \frac{\partial ^t \mathbf{x}/ \partial r_i} are the covariant base vectors, then the Green-Lagrange strain components can be written as:
+
+\begin{equation}
+\tilde{\epsilon}_{ij} = \frac{1}{2}(^t \mathbf{g_i} \cdot ^t \mathbf{g_j} - ^0 \mathbf{g_i} \cdot ^0 \mathbf{g_j})
+\end{equation}
+
+The expressions for the small and large strains in the 11, 22, 12, 13 and 23 directions are provided in equations 21 to 24 of [!cite](dvorkin1984continuum). The expressions for the shear strains in the 13 and 23 directions also include a correction for shear locking. In Moose the small strain increments are computed in
+[ADComputeIncrementalShellStrain](/ADComputeIncrementalShellStrain.md) and the small + large strain increments are computed in [ADComputeFiniteShellStrain](/ADComputeFiniteShellStrain.md)). Note that strain in the 33 direction is not computed and is set to zero.
```

Comment:
Is that right about the strain in the 33 direction being 0?  I thought the stress was 0 and we computed the strain to give that.

Reply:
I have corrected this statement. We are not computing the strain in the 33 direction. Instead the elasticity tensor is condensed appropriately.